### ISSUE TITLE
PiT 24.2: offline screen only happens on route change

### ISSUE THREAD

**Issue by manolo** [vaadin/flow#17344]
### Description of the bug

When using a flow application, if I turn network to offline, the offline screen appears only in certain actions (routing) even I can see round trips to server.

![offline](https://github.com/vaadin/flow/assets/161853/4bcba485-4876-465c-ac09-576ac7e4e7a6)

### Expected behavior

offline screen should appear as soon as a round trip fails because a network connectivity issue

### Minimal reproducible example

1. Run the bookstore for flow app
```
git clone https://github.com/vaadin/bookstore-example.git
cd bookstore-example
mvn clean install -Pproduction
mvn -Pproduction jetty:run-war 
```
2. switch network to offline
3. click on any element on the grid, there are request to server but not offline screen
4. click on any link in the side menu, offline screen appears

### Versions

- Vaadin / Flow version: 24.1.4

**Comment by knoobie** [comment]
This sounds more user friendly to be honest 😅 instantly showing an offline page because the network was gone for a second sounds a bit drastic, staying on the page with the information on the top feels more natural and hopefully the application just recovers when the network is online again without any data loss from the current view.

**Comment by manolo** [comment]
That's true when there is not user interaction, but when the user performs an action, the offline screen should be shown.
There are ways for customising the offline screen, by for instance making it a semi-transparent layer, then the user cannot interact with the UI but still  the last info keeps visible

**Comment by mshabarov** [comment]
Tried with different versions and looks like this is how V23.3 and V24.0 behave as well.

Flow fallbacks to the offline page when only navigation action is done by user, service worker does this with `workbox` as coded here https://github.com/vaadin/flow/blob/b965e91e8bfb00e69b3ffe8aff804e0eb9665285/flow-server/src/main/resources/com/vaadin/flow/server/frontend/sw.ts#L92

For "non-navigation" events like clicking of UI component or inputing data, there is nothing in the service working, nor in the PWA-related Flow codes, where such a fallback is registered.

It seems to me that this behaviour is a by-design. I agree with @knoobie that showing an offline page upon non-navigation interactions is a worse user experience, and, in my opinion is excessive - "connection lost" indicator is enough for users to see that the app is offline. Personally, I'd prefer to keep having the data on the page even without any semi-transparent overlay.

I'd propose to not consider this as a blocker for V24.2 and even not a bug in Flow.
We have to understand how many users would like to see/show the offline page always.
Before that information having on our hands, I would propose to do nothing within this issue.

**Comment by manolo** [comment]
Ok, thank @mshabarov, moved to non-block in PiT, I'll keep open for some time, to see if there are more opinions/thoughts
